### PR TITLE
Further AWS deployment improvements

### PR DESCRIPTION
**Pull request by ttmc**
This pull request adds a bunch of new stuff to the AWS deployment scripts:
* An easier way to make Fabric send the `bigchaindb init` command to only one node: use the `@hosts(hostname)` decorator.
* The `make_confiles.sh` script to generate a set of default BigchainDB config files. You have to do this before calling `./startup.sh` as explained in the updated documentation for AWS deployment.
* The `clusterize_confiles.py` script to make all the config files for a cluster consistent with each other and with the hostnames of the nodes in the cluster
* Updated `startup.sh` and `fabfile.py` to clusterize the set of config files and send a different config file to each node in the cluster

I also figured out a way to make sure the nodes are all ready to start getting commands from Fabric (via ssh), rather than waiting 45 seconds. I just create a local socket and then try to connect it to port 22 on all the nodes, and keep trying those connections until they all work.

Note: These scripts are just to help us get a test cluster up and running now, before we implement a better way of handling the list of public keys in the federation, like what's outlined in Issue #187.

### CODE REVIEW

File: deploy-cluster-aws/clusterize_confiles.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        pubkeys.append(pubkey)
+
+# Rewrite each config file, one at a time
+for i, filename in enumerate(conf_files):
+    file_path = os.path.join(conf_dir, filename)
+    with open(file_path, 'r') as f:
+        conf_dict = json.load(f)
+        # The keyring is the list of *all* public keys
+        # minus the config file's own public key
+        keyring = list(pubkeys)
```
Comment (r-marques) [inline deploy-cluster-aws/clusterize_confiles.py]: smart

File: deploy-cluster-aws/fabfile.py [hunk 2]
```diff
@@ -138,13 +133,26 @@ def configure_bigchaindb():
...
 
+# Send the specified configuration file to
+# the remote host and save it there in
+# ~/.bigchaindb
+# Use in conjunction with set_host()
+# No @parallel
+@task
+def send_confile(confile):
+    put('confiles/' + confile, 'tempfile')
+    sudo('mv tempfile ~/.bigchaindb')
```
Comment (r-marques) [inline deploy-cluster-aws/fabfile.py]: do we need sudo here?
I am afraid that the bigchaindb process may not have the right read/write permissions

**Comment by r-marques** [comment]
some comments. other then that seems :+1:

File: deploy-cluster-aws/fabfile.py (hunk 2 above)
Comment (ttmc) [inline deploy-cluster-aws/fabfile.py]: Good question. Originally, I tried to do something like `put('confiles/' + confile, '~/.bigchaindb')` but I was getting an error about not having sufficient permissions. So I just did it in two steps, using sudo to move the file into ~.

I was curious to see if sudo() was necessary, using run() instead. It works with run()!

I'll push the new code that uses run() instead.

**Pull request merged**